test/cctest/test_node_debug_support.cc

```diff
@@ -0,0 +1,127 @@
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  v8::Local<v8::Object> object = v8::Object::New(isolate_);
+  node::BaseObject *obj = new node::BaseObject(*env, object);
+
+  auto expected = (uintptr_t)&(obj->persistent());
+  auto calculated = (uintptr_t)((void*)obj) + nodedbg_class__BaseObject__persistent_handle;
+  EXPECT_EQ(expected, calculated);
+}
+
+
+TEST_F(DebugSymbolsTest, EnvironmentHandleWrapQueue) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  auto expected = (uintptr_t)((*env)->handle_wrap_queue());
+  auto calculated = ((uintptr_t)(*env)) + + nodedbg_class__Environment__handleWrapQueue;
+  EXPECT_EQ(expected, calculated);
+}
+
+TEST_F(DebugSymbolsTest, EnvironmentReqWrapQueue) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  auto expected = (uintptr_t)((*env)->req_wrap_queue());
+  auto calculated = ((uintptr_t)(*env)) + + nodedbg_class__Environment__reqWrapQueue;
+  EXPECT_EQ(expected, calculated);
+}
+
+TEST_F(DebugSymbolsTest, HandleWrapList) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  uv_handle_t handle_;
+
+  node::tracing::TraceEventHelper::SetTracingController(
+    new v8::TracingController());
+
+  auto obj_template = v8::FunctionTemplate::New(isolate_);
+  obj_template->InstanceTemplate()->SetInternalFieldCount(1);
+
+  v8::Local<v8::Object> object = obj_template->GetFunction()->NewInstance(env.context_).ToLocalChecked();
+  auto *obj = new TestHandleWrap((*env), object, &handle_);
+
+  auto queue = (uintptr_t)(*env)->handle_wrap_queue();
+  auto head = queue + nodedbg_class__HandleWrapQueue__headOffset;
+  auto next = head + nodedbg_class__HandleWrapQueue__nextOffset;
+  next = *reinterpret_cast<uintptr_t *>(next);
+
+  auto expected = (uintptr_t)obj;
+  auto calculated = next - nodedbg_class__HandleWrap__list;
+  EXPECT_EQ(expected, calculated);
+}
+
+TEST_F(DebugSymbolsTest, ReqWrapList) {
+  const v8::HandleScope handle_scope(isolate_);
+  const Argv argv;
+  Env env {handle_scope, isolate_, argv, this};
+
+  node::tracing::TraceEventHelper::SetTracingController(
+    new v8::TracingController());
+
+  auto obj_template = v8::FunctionTemplate::New(isolate_);
+  obj_template->InstanceTemplate()->SetInternalFieldCount(1);
+
+  v8::Local<v8::Object> object = obj_template->GetFunction()->NewInstance(env.context_).ToLocalChecked();
+  auto *obj = new TestReqWrap((*env), object);
+
+  auto queue = (uintptr_t)(*env)->req_wrap_queue();
+  auto head = queue + nodedbg_class__ReqWrapQueue__headOffset;
+  auto next = head + nodedbg_class__ReqWrapQueue__nextOffset;
+  next = *reinterpret_cast<uintptr_t *>(next);
+
+  auto expected = (uintptr_t)obj;
+  auto calculated = next - nodedbg_class__ReqWrap__node;
+  EXPECT_EQ(expected, calculated);
+}
```

Comment:
Please check with valgrind that the tests don't leak memory.

Reply:
Valgrind flagged one memory leak on `TEST_F(DebugSymbolsTest, BaseObjectPersistentHandle)`, which is the one you mentioned above.